### PR TITLE
SALTO-3553/block removal of active schemes

### PR DESCRIPTION
**Pull request by guyhamo**
add change validator to block removal of schemes that are in use by a project

---

_Additional context for reviewer_
added tests - 100% covered
---
_Release Notes_: 
jira_adapter:
* add change validator to block removal of schemes that are in use by proejects

---
_User Notifications_:

### CODE REVIEW

File: packages/jira-adapter/test/change_validators/active_workflow_schemes.test.ts [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+import { toChange, ObjectType, ElemID, InstanceElement, ReferenceExpression, ReadOnlyElementsSource } from '@salto-io/adapter-api'
+import { buildElementsSourceFromElements } from '@salto-io/adapter-utils'
+import { JIRA } from '../../src/constants'
+import { activeWorkflowSchemeChangeValidator } from '../../src/change_validators/active_workflow_schemes'
+
+describe('workflow scheme migration', () => {
```
Comment (guyhamo) [inline packages/jira-adapter/test/change_validators/active_workflow_schemes.test.ts:21]: fix name

File: packages/jira-adapter/src/change_validators/active_workflow_scheme_deletion.ts [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+  elemID,
+  severity: 'Error' as SeverityLevel,
+  message: 'Can’t remove schemes that are being used',
+  detailedMessage: `This scheme is currently used by ${projects.length === 1 ? 'project' : 'projects'} ${projects.map(project => project.elemID.name).join(', ')}, and can’t be deleted`,
+})
+
+export const activeWorkflowSchemeDeletionValidator: ChangeValidator = async (changes, elementSource) => {
+  const relevantChanges = changes
+    .filter(isRemovalChange)
+    .filter(change => getChangeData(change).elemID.typeName === WORKFLOW_SCHEME_TYPE_NAME)
```
Comment (alonstern) [inline packages/jira-adapter/src/change_validators/active_workflow_scheme_deletion.ts:35]: The ticket is not specifically about workflow scheme, it is about all the schemes in the project

Comment (guyhamo) [reply packages/jira-adapter/src/change_validators/active_workflow_scheme_deletion.ts:35]: oh, ok I will fix that

File: packages/jira-adapter/src/change_validators/active_scheme_deletion.ts [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+const { isDefined } = values
+
+const SCHEME_TYPE_TO_PROJECT_FIELD: Record<string, string> = {
+  [WORKFLOW_SCHEME_TYPE_NAME]: 'workflowScheme',
+  [PERMISSION_SCHEME_TYPE_NAME]: 'permissionScheme',
+  [NOTIFICATION_SCHEME_TYPE_NAME]: 'notificationScheme',
+  [ISSUE_TYPE_SCHEMA_NAME]: 'issueTypeScheme',
+  IssueTypeScreenScheme: 'issueTypeScreenScheme',
+  ScreenScheme: 'screenScheme',
+  [SECURITY_SCHEME_TYPE]: 'issueSecurityScheme',
```
Comment (alonstern) [inline packages/jira-adapter/src/change_validators/active_scheme_deletion.ts:30]: I think `FieldConfigScheme` and `PriorityScheme` are missing here (`PriorityScheme` exist only in DC`

File: packages/jira-adapter/src/change_validators/workflow_scheme_migration.ts [hunk 4]
```diff
@@ -48,7 +48,7 @@ export type StatusMigration = {
   newStatusId?: ReferenceExpression
 }
 
-const projectHasWorkflowSchemeReference = (project: InstanceElement): boolean =>
+export const projectHasWorkflowSchemeReference = (project: InstanceElement): boolean =>
```
Comment (alonstern) [inline packages/jira-adapter/src/change_validators/workflow_scheme_migration.ts:51]: nit: I think the export is not needed anymore

**Review by alonstern — APPROVED** [review]
👌

**Pull request merged**